### PR TITLE
Refactor/thr momentum management

### LINKED ISSUE #304: Convert thrMomentumManagement to C++
**Describe your use case**
The ``thrMomentumManagement`` module will have new features, but it needs to be converted to C++ first.

**Describe alternatives solutions you've considered**
Maintained all functionality but the module is now written in C++.

**Additional context**
N/A.

### PR DESCRIPTION
**Pull request by joaogvcarneiro**
* **Tickets addressed:** #304 
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

This PR recreates the ``thrMomentumManagement`` module in C++ for further development. The first commit adds the module files, the second adds the unit test and the last adds documentation.

## Verification

Test files were copied from the C module except for some minor adjustments.

## Documentation

Ported the old module documentation into ``rst`` format.

## Future work

This conversion is intended to be used in future additions.

### CODE REVIEW

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#include "architecture/messaging/messaging.h"
+
+#include "architecture/msgPayloadDefC/RWArrayConfigMsgPayload.h"
+#include "architecture/msgPayloadDefC/RWSpeedMsgPayload.h"
+#include "architecture/msgPayloadDefC/CmdTorqueBodyMsgPayload.h"
+
+/*! @brief Module configuration message definition. */
+class ThrMomentumManagementCpp: public SysModel {
+public:
+    ThrMomentumManagementCpp() = default;                                   //!< Constructor
```
Comment (patkenneally) [inline src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h]: :small_red_triangle_down: You can delete the constructor/destructor given the compiler will generate a default  for each.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#include "fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h"
+#include "architecture/utilities/linearAlgebra.h"
+#include "architecture/utilities/avsEigenSupport.h"
+#include <Eigen/Core>
+
+void ThrMomentumManagementCpp::Reset(uint64_t currentSimNanos)
+{
+    // Check if the required input messages are included
+    if (!this->rwConfigDataInMsg.isLinked()) {
+        this->bskLogger->bskLog(BSK_ERROR, "Error: thrMomentumManagementCpp.rwConfigDataInMsg wasn't connected.");
```
Comment (patkenneally) [inline src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.cpp]: :small_red_triangle_down: "Error: " is redundant in these log strings as the `BSK_ERROR` toggle prepends "BSK_ERROR" already.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.i [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+%}
+%include "std_string.i"
+%include "swig_eigen.i"
+%include "swig_conly_data.i"
+
+%include "sys_model.i"
+%include "thrMomentumManagementCpp.h"
+
+%include "architecture/msgPayloadDefC/RWArrayConfigMsgPayload.h"
+struct RWArrayConfigMsg_C;
```
Comment (patkenneally) [inline src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.i]: :small_red_triangle_down: These `struct` directives are no longer needed and can be removed.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+# Import all of the modules that we are going to be called in this simulation
+from Basilisk.utilities import SimulationBaseClass, macros, fswSetupRW
+from Basilisk.fswAlgorithms import thrMomentumManagementCpp
+from Basilisk.architecture import messaging
+
+
+@pytest.mark.parametrize("hsMinCheck", [(0), (1)])
+def test_thrMomentumManagement(show_plots, hsMinCheck):
+    unitTaskName = "unitTask"               # arbitrary name (don't change)
```
Comment (patkenneally) [inline src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py]: :large_blue_circle: There are many redundant comments throughout the unit tests in the code base. We should seek to clean them out as we go.

**Review by patkenneally — CHANGES_REQUESTED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h (hunk 1 above)
Comment (joaogvcarneiro) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h]: Done.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.cpp (hunk 2 above)
Comment (joaogvcarneiro) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.cpp]: Done.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.i (hunk 3 above)
Comment (joaogvcarneiro) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.i]: Done.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py (hunk 4 above)
Comment (joaogvcarneiro) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py]: Done.

**Review by patkenneally — APPROVED** [review]

**Pull request merged**